src/__test__/bitDepth1.write.test.ts

```diff
@@ -173,6 +176,9 @@ describe('encode image with bitDepth of 1', () => {
       bitDepth: 1,
       components: 1,
       channels: 1,
+      compression: 0,
+      colorMasks: [16711680, 65280, 255],
+      logicalColorSpace: 0,
```

Comment:
What is the difference between 0 and 2?

Reply:
When you read 1 bit images, you check the value is 2, but you write them, you write 0.
So the question is what is the difference according to the spec and are both tests correct?

Reply:
It was supposed to represent the number of colors used in an image. However, the way it was initially encoded was that the number of colors used was the number of possible colors. So, in most cases for 1-bit images it should be 2. If you have an all-white or all-black image it should be 1 (to be tested). If the bitDepth is >= 8, the number can be set to 0.

Reply:
I didn't find much data on it in the internet, but AI tells me that it can be used by applications to perform more efficient color quantization when converting the image to a format with a lower bit depth. So, it's used for optimization, I don't think it's meaningful in basic cases. I can make an algorithm that calculates the number of colors used, but if you think it's an overkill, I would just remove it